src/javascript/app/pages/trade/markets/markets-dropdown.jsx

```diff
@@ -0,0 +1,323 @@
+                const offsetTop = div.offsetTop - container.scrollTop - paddingOffset;
+
+                if (offsetTop <= 0 && Math.abs(offsetTop) < Math.abs(closestOffset)) {
+                    closestOffset = offsetTop;
+                    closestMarket = div.getAttribute('data-id');
+                }
+            });
+    
+            if (closestMarket) {
+                setActiveMarket(closestMarket);
+            }
+        };
+    
+        const handleScroll = () => {
+            if (!isScrolling.current) {
+                checkActiveMarket();
+            }
+        };
+    
+        container.addEventListener('scroll', handleScroll);
+    
+        return () => {
+            container.removeEventListener('scroll', handleScroll);
+        };
+    }, [isMounted]);
+    
+    const getactiveMarketIndex = () => Object.keys(markets).indexOf(activeMarket);
+
+    const scrollToMarketByIndex = (index) => {
+        isScrolling.current = true;
+        const container = itemsContainer.current;
+
+        if (container) {
+            const marketDivs = container ? container.children : [];
+            if (marketDivs[index]) {
+                const offsetTop = marketDivs[index].offsetTop - container.offsetTop;
+                container.scrollTo({
+                    top     : offsetTop,
+                    behavior: 'smooth',
+                });
+            }
+            setTimeout(() => {
+                isScrolling.current = false;
+            }, 1000);
+        }
+    };
+
+    const handleUnderlyingClick = (underlying) => {
+        Defaults.set(UNDERLYING, underlying);
+        setSelectedMarket(underlying);
+        triggerMarketChange();
+
+        // Trigger the old form to enable other required effects from it
+        document.querySelectorAll('.symbol_name').forEach(el => el.id === underlying && el.click());
+
+        closeMarketDropdown();
+    };
+
+    useEffect(() => {
+        if (isMounted && itemsContainer.current) {
+            const paddingOffset = 96;
+            const container = itemsContainer.current;
+            const selectedElement = container.querySelector('.market-item-selected');
+            if (selectedElement) {
+                const offsetTop = selectedElement.offsetTop - container.offsetTop - paddingOffset;
+                container.scrollTo({
+                    top     : offsetTop,
+                    behavior: 'auto',
+                });
+            }
+                
+        }
+    
+        const toActiveMarket = () => {
+            const container = itemsContainer.current;
+            let selectedElement = container.querySelector('.market-item-selected');
+        
+            while (selectedElement && selectedElement !== document) {
+                if (selectedElement.tagName === 'DIV' && selectedElement.hasAttribute('data-id')) {
+                    const market =  selectedElement.getAttribute('data-id');
+                    return setActiveMarket(market);
```

Comment:
```suggestion
                    setActiveMarket(market);
```
Do we need to return anything here? it looks like we just want to set the market.
Also, can we please give `toActiveMarket` another name like `goToActiveMarket`, or `selectActiveMarket`, or anything else that clarifies its purpose?🙏🏻
Or since `toActiveMarket` is used only once, we can also place its code where it's executed instead of declaring it as a function.